_posts/2022-01-20-making-github-action.md

```diff
@@ -0,0 +1,131 @@
+
+```
+name: 'XLSX to CSVs'
+description: "Convert an XLSX file's sheets to CSVs"
+inputs:
+  input_file:
+    description: 'xlsx file or directory to convert'
+    required: true
+    default: 'test.xlsx'
+runs:
+  using: 'docker'
+  image: 'Dockerfile'
+  args:
+    - ${{ inputs.input_file }}
+```
+
+This is a fairly minimal example of an action.yml file. More [syntax options](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions) are available.
+
+Finally, to make the Action available to Workflows, we need commit and push the action to GitHub and then [make a named release](https://docs.github.com/en/articles/creating-releases). Now Workflows can call this action by referring to its `owner/action_name@release`. Note that it is possible to create the action within the repository that you want to run the workflow in, but if you want to make this Action discoverable to others by [publishing it in the GitHub Marketplace](https://docs.github.com/en/actions/creating-actions/publishing-actions-in-github-marketplace), you'll need to keep it in its own public repository.
+
+We can now use this Action in a Workflow: for example, we want to convert all pushed xlsx files to csv files in a repository and then commit those changes. We'll define this in the repository in a yml file located in `.github/workflows/`. This metadata file defines the [triggering event](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows) and breaks down the series of jobs to be performed on that event. Each job defines the environment that the Actions will run in, then lists the actions to be taken, along with their inputs and outputs. Note here that we are using two other Actions that are from the GitHub Marketplace to do the checkout and commit steps, and that we can take advantage of the [environment variables](https://docs.github.com/en/actions/learn-github-actions/environment-variables) defined for the Workflow context.
+
+```
+on: [push]
+
+jobs:
+  convert-csvs:
+    runs-on: ubuntu-latest
+    name: Convert xlsx files to csvs
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Convert to CSV
+        uses: CanDIG/xlsx2csv-action@v2.0
+        id: convert
+        with:
+          input_file: .
+      - name: Commit changes
+        uses: EndBug/add-and-commit@v4
+        with:
+          author_name: xlsx2csv bot
+          message: "convert xlsx files to csvs"
+          add: "*"
+          ref: ${{env.GITHUB_REF}}
+```
+
+Once this workflow metadata file is pushed to GitHub, this workflow will be executed every time the triggering event occurs. Confusingly, you can see the workflow runs in the Actions tab of your repository, e.g. https://github.com/CanDIG/xlsx2csv-action/actions.
```

Comment:
Can you make `https://github.com/CanDIG/xlsx2csv-action/actions` a markdown link?